src/client/Microsoft.Identity.Client/AppConfig/ExternalTokenResult.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Token result from external token provider
+    /// </summary>
+    public class ExternalTokenResult
+    {
+        /// <summary>
+        /// Access token
+        /// Mandatory
+        /// </summary>
+        public string RawAccessToken { get; set; }
+
+        /// <summary>
+        /// Tenant Id for client application
+        /// </summary>
+        public string TenantId { get; set; }
```

Comment:
Is there a case where the result.tenant_id is different than input.tenant_id? I think in client_credentials we always treat it as the same tenant, because ESTS does not tell us if the token is for a different tenant. This is a user scenario only, where we derive this info from idtoken `tid` claim.